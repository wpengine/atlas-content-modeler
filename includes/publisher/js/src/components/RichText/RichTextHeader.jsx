import React from "react";

const RichTextHeader = ({ modelSlug, field }) => {
	return (
		<>
			<label
				htmlFor={`atlas-content-modeler[${modelSlug}][${field.slug}]`}
			>
				{field.name}
			</label>
			{field?.description && (
				<p className="help mb-4">{field.description}</p>
			)}
		</>
	);
};

export default RichTextHeader;
