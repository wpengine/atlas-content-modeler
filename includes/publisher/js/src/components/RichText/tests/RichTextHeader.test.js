import React from "react";
import renderer from "react-test-renderer";
import RichTextHeader from "../RichTextHeader";

describe("RichTextHeader", () => {
	const props = {
		modelSlug: "cats",
		field: {
			slug: "name",
			name: "Name",
			description: "The cat's real name.",
			value: "Kahn The Destroyer",
		},
	};
	const TestComponent = renderer.create(<RichTextHeader {...props} />);

	it("renders a matching snapshot", () => {
		expect(TestComponent.toJSON()).toMatchSnapshot();
	});
});
