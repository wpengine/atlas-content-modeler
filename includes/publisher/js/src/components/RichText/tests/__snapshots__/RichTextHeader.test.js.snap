// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RichTextHeader renders a matching snapshot 1`] = `
Array [
  <label
    htmlFor="atlas-content-modeler[cats][name]"
  >
    Name
  </label>,
  <p
    className="help mb-4"
  >
    The cat's real name.
  </p>,
]
`;
